src/uct/ib/mlx5/ib_mlx5_dv.c

```diff
@@ -50,24 +54,63 @@ static ucs_status_t uct_ib_mlx5_device_init(uct_ib_device_t *dev)
         goto err_cq;
     }
 
+    srq_attr.attr.max_sge   = 1;
+    srq_attr.attr.max_wr    = 1;
+    srq = ibv_create_srq(pd, &srq_attr);
+    if (srq == NULL) {
+        ucs_error("ibv_create_cq() failed: %m");
+        status = UCS_ERR_IO_ERROR;
+        goto err_srq;
+    }
+
     qp_attr.send_cq              = cq;
     qp_attr.recv_cq              = cq;
-    qp_attr.cap.max_send_wr      = 1;
-    qp_attr.cap.max_send_sge     = 1;
     qp_attr.qp_type              = IBV_QPT_DRIVER;
     qp_attr.comp_mask            = IBV_QP_INIT_ATTR_PD;
     qp_attr.pd                   = pd;
+    qp_attr.srq                  = srq;
 
     dv_attr.comp_mask            = MLX5DV_QP_INIT_ATTR_MASK_DC;
-    dv_attr.dc_init_attr.dc_type = MLX5DV_DCTYPE_DCI;
+    dv_attr.dc_init_attr.dc_type = MLX5DV_DCTYPE_DCT;
+    dv_attr.dc_init_attr.dct_access_key = UCT_IB_KEY;
 
-    /* create DCI qp successful means DC is supported */
+    /* create DCT qp successful means DC is supported */
     qp = mlx5dv_create_qp(ctx, &qp_attr, &dv_attr);
-    if (qp) {
-        ibv_destroy_qp(qp);
+    if (qp == NULL) {
+        goto err_qp;
+    }
+
```

Comment:
is not creating a qp enough to check DC support?
Why is it required to move it to RTR state?

Reply:
Move to RTR state triggers in kernel actual CREATE_DCT